octodns/record/svcb.py

```diff
@@ -0,0 +1,251 @@
+#
+# This file describes the SVCB and HTTPS records as defined in RFC 9460
+# It also supports the 'ech' SvcParam as defined in draft-ietf-tls-svcb-ech-02
+#
+
+from base64 import b64decode
+from binascii import Error as binascii_error
+from ipaddress import AddressValueError, IPv4Address, IPv6Address
+
+from fqdn import FQDN
+
+from ..equality import EqualityTupleMixin
+from ..idna import idna_encode
+from .base import Record, ValuesMixin, unquote
+from .chunked import _ChunkedValue
+from .rr import RrParseError
+
+SUPPORTED_PARAMS = {}
+
+
+def validate_svcparam_port(svcparamvalue):
+    reasons = []
+    try:
+        port = int(svcparamvalue)
+        if 0 < port > 65535:
+            reasons.append(f'port {port} is not a valid number')
+    except ValueError:
+        reasons.append('port is not a number')
+    return reasons
+
+
+def validate_svcparam_alpn(svcparamvalue):
+    reasons = []
+    alpns = svcparamvalue.split(',')
+    for alpn in alpns:
+        reasons += _ChunkedValue.validate(alpn, 'SVCB')
+    return reasons
+
+
+def validate_svcparam_iphint(ip_version, svcparamvalue):
+    reasons = []
+    addresses = svcparamvalue.split(',')
+    for address in addresses:
+        try:
+            if ip_version == 4:
+                IPv4Address(address)
+            if ip_version == 6:
+                IPv6Address(address)
+        except AddressValueError:
+            reasons.append(
+                f'ip{ip_version}hint "{address}" is not a valid IPv{ip_version} address'
+            )
+    return reasons
+
+
+def validate_svcparam_ip4hint(svcparamvalue):
+    return validate_svcparam_iphint(4, svcparamvalue)
+
+
+def validate_svcparam_ip6hint(svcparamvalue):
+    return validate_svcparam_iphint(6, svcparamvalue)
+
+
+def validate_svcparam_mandatory(svcparamvalue):
+    reasons = []
+    mandatories = svcparamvalue.split(',')
+    for mandatory in mandatories:
+        if (
+            mandatory not in SUPPORTED_PARAMS.keys()
+            and not mandatory.startswith('key')
+        ):
+            reasons.append(f'unsupported SvcParam "{mandatory}" in mandatory')
+        if mandatory.startswith('key'):
+            reasons += validate_svckey_number(mandatory)
+    return reasons
+
+
+def validate_svcparam_ech(svcparamvalue):
+    try:
+        b64decode(svcparamvalue, validate=True)
+    except binascii_error:
+        return ['ech SvcParam is invalid Base64']
+
+
+def validate_svckey_number(paramkey):
+    try:
+        paramkeynum = int(paramkey[3:])
+        if 7 < paramkeynum > 65535:
+            return [f'SvcParam key "{paramkey}" has wrong key number']
+    except ValueError:
+        return [f'SvcParam key "{paramkey}" has wrong format']
+    return []
+
+
+SUPPORTED_PARAMS = {
```

Comment:
Would be good to link this to https://datatracker.ietf.org/doc/html/rfc9460#keys since that's the source of the keys. Al

```suggestion
# cc https://datatracker.ietf.org/doc/html/rfc9460#keys
SUPPORTED_PARAMS = {
```